### PR TITLE
Partially change the BP entry flows to be synchronous

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/1275/add-performance-instrumentation-for-bp-entry-flow

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryEffectHandlerTest.kt [hunk 1]
```diff
@@ -4,26 +4,37 @@ import com.nhaarman.mockitokotlin2.mock
...
       bloodPressureRepository = mock(),
       appointmentsRepository = mock(),
       userClock = userClock,
-      schedulersProvider = TrampolineSchedulersProvider(),
-      uuidGenerator = mock()
-  ).build()
+      schedulersProvider = TestSchedulersProvider.trampoline(),
+      uuidGenerator = mock(),
+      currentUser = { user }
+  ) { facility }.build()
```
Comment (msasikanth) [inline app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryEffectHandlerTest.kt:36]: Can we move this inside the constructor, please? (or is there a reason it's trailing lambda?)

File: app/src/test/java/org/simple/clinic/bp/entry/BloodPressureValidationMockDateValidatorTest.kt [hunk 2]
```diff
@@ -186,15 +179,14 @@ class BloodPressureValidationMockDateValidatorTest {
...
         bloodPressureRepository = bloodPressureRepository,
         appointmentsRepository = appointmentRepository,
         userClock = testUserClock,
-        schedulersProvider = TrampolineSchedulersProvider(),
-        uuidGenerator = FakeUuidGenerator.fixed(UUID.fromString("7283abf4-b718-4379-b101-46f011b5536b"))
-    ).build()
+        schedulersProvider = TestSchedulersProvider.trampoline(),
+        uuidGenerator = FakeUuidGenerator.fixed(UUID.fromString("7283abf4-b718-4379-b101-46f011b5536b")),
+        currentUser = { user }
+    ) { facility }.build()
```
Comment (msasikanth) [inline app/src/test/java/org/simple/clinic/bp/entry/BloodPressureValidationMockDateValidatorTest.kt:189]: Can we move this inside the constructor, please?

File: app/src/test/java/org/simple/clinic/bp/entry/BloodPressureValidationTest.kt [hunk 3]
```diff
@@ -184,15 +167,14 @@ class BloodPressureValidationTest {
...
         bloodPressureRepository = bloodPressureRepository,
         appointmentsRepository = appointmentRepository,
         userClock = testUserClock,
-        schedulersProvider = TrampolineSchedulersProvider(),
-        uuidGenerator = FakeUuidGenerator.fixed(UUID.fromString("abb04673-6ec0-4e1a-a4ad-5380e6f7e233"))
-    ).build()
+        schedulersProvider = TestSchedulersProvider.trampoline(),
+        uuidGenerator = FakeUuidGenerator.fixed(UUID.fromString("abb04673-6ec0-4e1a-a4ad-5380e6f7e233")),
+        currentUser = { user }
+    ) { facility }.build()
```
Comment (msasikanth) [inline app/src/test/java/org/simple/clinic/bp/entry/BloodPressureValidationTest.kt:177]: Same comment as above, moving this inside constructor.

**Review by msasikanth — CHANGES_REQUESTED** [review]
PR looks good overall, Left couple of comments PTAL

File: app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryEffectHandlerTest.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryEffectHandlerTest.kt:36]: Huh, I thought I had corrected this. Maybe I missed it. Will fix.

File: app/src/test/java/org/simple/clinic/bp/entry/BloodPressureValidationMockDateValidatorTest.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/bp/entry/BloodPressureValidationMockDateValidatorTest.kt:189]: Yup, moving it.

File: app/src/test/java/org/simple/clinic/bp/entry/BloodPressureValidationTest.kt (hunk 3 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/bp/entry/BloodPressureValidationTest.kt:177]: Oh wait, I think the IDE made this change automatically and I ended up correcting it for one of the test classes and not the others. 😅

**Review by msasikanth — APPROVED** [review]

**Pull request merged**